packages/features/bookings/Booker/components/OverlayCalendar/OverlayCalendarContainer.tsx

```diff
@@ -0,0 +1,136 @@
+import { useSession } from "next-auth/react";
+import { useRouter, useSearchParams, usePathname } from "next/navigation";
+import { useState, useCallback, useEffect } from "react";
+
+import dayjs from "@calcom/dayjs";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { trpc } from "@calcom/trpc/react";
+import { Button, Switch } from "@calcom/ui";
+import { Settings } from "@calcom/ui/components/icon";
+
+import { useBookerStore } from "../../store";
+import { OverlayCalendarContinueModal } from "../OverlayCalendar/OverlayCalendarContinueModal";
+import { OverlayCalendarSettingsModal } from "../OverlayCalendar/OverlayCalendarSettingsModal";
+import { useLocalSet } from "../hooks/useLocalSet";
+import { useOverlayCalendarStore } from "./store";
+
+export function OverlayCalendarContainer() {
+  const { t } = useLocale();
+  const [continueWithProvider, setContinueWithProvider] = useState(false);
+  const [calendarSettingsOverlay, setCalendarSettingsOverlay] = useState(false);
+  const { data: session } = useSession();
+  const setOverlayBusyDates = useOverlayCalendarStore((state) => state.setOverlayBusyDates);
+  const selectedDate = useBookerStore((state) => state.selectedDate);
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+
+  // Move this to a hook
+  const { set, clearSet } = useLocalSet<{
+    credentialId: number;
+    externalId: string;
+  }>("toggledConnectedCalendars", []);
+  const overlayCalendarQueryParam = searchParams.get("overlayCalendar");
+  const { data: overlayBusyDates } = trpc.viewer.availability.calendarOverlay.useQuery(
+    {
+      loggedInUsersTz: dayjs.tz.guess() || "Europe/London",
```

Comment:
Shouldn't we use the tz of the bookings page here? This may also related to the comment below about the timezone issue

Reply:
Youre 100% right good spot